### ISSUE TITLE
MasterElement inconsistency: `general_shape_fcn` not implemented for Wedge, Pyramids etc

### ISSUE THREAD

**Issue by sayerhs** [NaluCFD/Nalu#303]
The `general_shape_fcn` is only implemented for HEX in 3-D but not for other topologies. Instead there are `wedge_shape_fcn`, `pyr_shape_fcn` etc. This causes issues for overset in [overset assemble](https://github.com/NaluCFD/Nalu/blob/master/src/overset/AssembleOversetSolverConstaintAlgorithm.C#L175) when used with mixed topologies. I am able to work around this temporarily by the code snippet below (wedge for example).

```
+  void general_shape_fcn(
+    const int npts,
+    const double *par_coord,
+    double* shape_fcn)
+  { wedge_shape_fcn(npts, par_coord, shape_fcn); }
+
```

However, I am unsure of the reasons behind this. Was this just an oversight, or is there a specific reason why the code defines `wedge_shape_fcn` instead of overriding `general_shape_fcn`?

**Comment by spdomin** [comment]
When we implemented overset with our PSAAP-2 student a few summers ago, we only focused on hex/tet/quad/tri elements. The hybrid overset mesh drove this decision. As such, the general_shape_function method was not implemented for Wedges and pyramids.

As for your second question, in Nalu, we use master elements using our chosen isoparametric range. In most cases, this choice was different from the standard that isInElement uses. Therefore, there is a disconnect in isoparametric ranges. You must use a isoparametric range consistent with isInElement for general_shape_fcn. We would have to check to see if the wedge_shape_fcn and pyramid_shape_fcn conform to the isInElement standard. I think that they might.. If so, we could simply change the name for the internal wedge and pyramid shape functions.
